datasets/batch_api_pipeline.js

```diff
@@ -0,0 +1,87 @@
+import fs from "fs";
+import OpenAI from "openai";
+
+// Initialize OpenAI client with API key from environment variables.
+const openai = new OpenAI({
+  apiKey: process.env.OPENAI_API_KEY,
+});
+
+/**
+ * Returns a promise that resolves after the given delay in milliseconds.
+ * @param {number} ms - Milliseconds to delay.
+ */
+function delay(ms) {
+  return new Promise(resolve => setTimeout(resolve, ms));
+}
+
+async function main() {
+  try {
+    // --- Step 1: Upload file ---
+    console.log("Uploading file...");
+    const uploadResponse = await openai.files.create({
+      file: fs.createReadStream("datasets/qa_dataset.jsonl"),
+      purpose: "batch",
+    });
+    console.log("Upload response:", uploadResponse);
+
+    // Get the file id from the upload response.
+    const inputFileId = uploadResponse.id;
+    console.log("Input file id:", inputFileId);
+    if (!inputFileId) {
+      throw new Error("Failed to get file id from upload response.");
+    }
+    
+    // --- Step 2: Create batch ---
+    console.log("Creating batch...");
+    const batchResponse = await openai.batches.create({
+      input_file_id: inputFileId,
+      endpoint: "/v1/chat/completions",
+      completion_window: "24h"
+    });
+    console.log("Batch creation response:", batchResponse);
+    
+    // Get the batch id from the batch creation response.
+    const batchId = batchResponse.id;
+    console.log("Batch id:", batchId);
+    if (!batchId) {
+      throw new Error("Failed to get batch id from batch creation response.");
+    }
+    
+    // --- Step 3: Poll for batch status until output_file_id is available ---
+    let outputFileId = null;
+    let attempts = 0;
+    const maxAttempts = 120; // e.g., 120 attempts x 5 seconds = 10 minutes maximum.
```

Comment:
I think we're being a little too strict about only giving them 10 minutes to process the batch. They could be overloaded. Also, this should make sense with the 24h completion window or whatever window is set. So if polling every 10 minutes and we give them 24h, the number of checks should be 6 * 24 = 144.